pkgs/applications/video/obs-studio/wrapper.nix

```diff
@@ -8,11 +8,17 @@ symlinkJoin {
   nativeBuildInputs = [ makeWrapper ];
   paths = [ obs-studio ] ++ plugins;
 
-  postBuild = ''
-    wrapProgram $out/bin/obs \
-      --set OBS_PLUGINS_PATH      "$out/lib/obs-plugins" \
-      --set OBS_PLUGINS_DATA_PATH "$out/share/obs/obs-plugins"
-  '';
+  postBuild =
+    let
+      # Some plugins needs extra environment, see obs-gstreamer for an example.
+      pluginArguments =
+        lib.lists.concatMap (plugin: plugin.obsWrapperArguments) plugins;
+    in ''
+      wrapProgram $out/bin/obs \
+        --set OBS_PLUGINS_PATH      "$out/lib/obs-plugins" \
+        --set OBS_PLUGINS_DATA_PATH "$out/share/obs/obs-plugins" \
```

Comment:
This looks kinda dangerous, what happens when `pluginArguments` is empty?

Probably better to use `concatStringsSep " " [ "--set OBS_PLUGINS_PATH      "$out/lib/obs-plugins" ]` instead.

Reply:
Also, using `concatStringsSep` means you don't need to call `toString pluginArguments`, instead you can just concat it to the other list of arguments.